ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/TimeoutParticipantTestContext.scala

```diff
@@ -0,0 +1,497 @@
+    s"$n checkpoints from offset $from for parties $parties",
+    delegate.checkpoints(n, from)(parties: _*),
+  )
+  override def firstCheckpoint(request: CompletionStreamRequest): Future[Checkpoint] = withTimeout(
+    s"First checkpoint for request $request",
+    delegate.firstCheckpoint(request),
+  )
+  override def firstCheckpoint(parties: Primitive.Party*): Future[Checkpoint] = withTimeout(
+    s"First checkpoint for parties $parties",
+    delegate.firstCheckpoint(parties: _*),
+  )
+  override def nextCheckpoint(request: CompletionStreamRequest): Future[Checkpoint] = withTimeout(
+    s"Next checkpoint for request $request",
+    delegate.nextCheckpoint(request),
+  )
+  override def nextCheckpoint(from: LedgerOffset, parties: Primitive.Party*): Future[Checkpoint] =
+    withTimeout(
+      s"Next checkpoint from offset $from for parties $parties",
+      delegate.nextCheckpoint(from, parties: _*),
+    )
+  override def configuration(overrideLedgerId: Option[String]): Future[LedgerConfiguration] =
+    withTimeout(
+      s"Configuration for ledger $overrideLedgerId",
+      delegate.configuration(overrideLedgerId),
+    )
+  override def checkHealth(): Future[HealthCheckResponse] =
+    withTimeout("Check health", delegate.checkHealth())
+  override def watchHealth(): Future[Seq[HealthCheckResponse]] =
+    withTimeout("Watch health", delegate.watchHealth())
+  override def getTimeModel(): Future[GetTimeModelResponse] =
+    withTimeout("Get time model", delegate.getTimeModel())
+  override def setTimeModel(
+      mrt: Instant,
+      generation: Long,
+      newTimeModel: TimeModel,
+  ): Future[SetTimeModelResponse] = withTimeout(
+    s"Set time model with mrt $mrt, generation $generation and new model $newTimeModel",
+    delegate.setTimeModel(mrt, generation, newTimeModel),
+  )
+  override def setTimeModelRequest(
+      mrt: Instant,
+      generation: Long,
+      newTimeModel: TimeModel,
+  ): SetTimeModelRequest = delegate.setTimeModelRequest(mrt, generation, newTimeModel)
+  override def setTimeModel(request: SetTimeModelRequest): Future[SetTimeModelResponse] =
+    withTimeout(
+      s"Set time model for request $request",
+      delegate.setTimeModel(request),
+    )
+  override private[infrastructure] def preallocateParties(
+      n: Int,
+      participants: Iterable[ParticipantTestContext],
+  ) = withTimeout(
+    s"Preallocate $n parties on participants ${participants.map(_.ledgerEndpoint)}",
+    delegate.preallocateParties(n, participants),
+  )
+  override def prune(
+      pruneUpTo: LedgerOffset,
+      attempts: Int,
+      pruneAllDivulgedContracts: Boolean,
+  ): Future[PruneResponse] = withTimeout(
+    s"Prune up to $pruneUpTo, with $attempts attempts and divulged contracts [$pruneAllDivulgedContracts]",
+    delegate.prune(pruneUpTo, attempts, pruneAllDivulgedContracts),
+  )
+
+  private def withTimeout[T](hint: String, future: => Future[T]): Future[T] = {
```

Comment:
What about existing custom timeouts in specific tests (if there are any)? Are they going to interfere with this one somehow?